scripts/tests/test_zkai_d128_block_receipt_composition_gate.py

```
@@ -0,0 +1,262 @@
+            json_path = tmp / "block-receipt.json"
+            tsv_path = tmp / "block-receipt.tsv"
+            COMPOSITION.write_outputs(payload, json_path, tsv_path)
+            loaded = json.loads(json_path.read_text(encoding="utf-8"))
+            self.assertEqual(loaded, payload)
+            self.assertIn("missing_bridge_slice", tsv_path.read_text(encoding="utf-8"))
+
+    def test_write_outputs_rejects_paths_outside_repo(self) -> None:
+        payload = self.fresh_payload()
+        with tempfile.TemporaryDirectory() as raw_tmp:
+            out = pathlib.Path(raw_tmp) / "outside.json"
+            with self.assertRaisesRegex(COMPOSITION.D128BlockReceiptError, "escapes repository"):
```

Ruff (0.15.12)

[warning] Use `pytest.raises` instead of unittest-style `assertRaisesRegex`

Replace `assertRaisesRegex` with `pytest.raises`

(PT027)